File: src/models/ProgrammeMobilite.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const ProgrammeMobilite = sequelize.define('ProgrammeMobilite', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  titre: {
    type: DataTypes.STRING(255),
    allowNull: false,
    defaultValue: 'Programme de Mobilité Internationale',
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: false,
    defaultValue: 'Description du programme de mobilité...',
  },
  criteres_eligibilite: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  image_couverture: {
    type: DataTypes.STRING(500),
    allowNull: true,
    comment: 'Chemin vers l\'image de couverture du programme de mobilité',
  },
  statut: {
    type: DataTypes.ENUM('actif', 'inactif'),
    defaultValue: 'actif',
  },
}, {
  tableName: 'programme_mobilite',
  timestamps: true,
});

module.exports = ProgrammeMobilite;

File: flutter_application_1/lib/models/programme_mobilite.dart
class ProgrammeMobilite {
  final int id;
  final String titre;
  final String description;
  final String? criteresEligibilite;
  final String? imageCouverture;
  final String statut; // actif | inactif

  ProgrammeMobilite({
    required this.id,
    required this.titre,
    required this.description,
    this.criteresEligibilite,
    this.imageCouverture,
    required this.statut,
  });

  bool get estActif => statut == 'actif';

  factory ProgrammeMobilite.fromJson(Map<String, dynamic> json) {
    return ProgrammeMobilite(
      id:                   json['id'],
      titre:                json['titre'],
      description:          json['description'],
      criteresEligibilite:  json['criteres_eligibilite'],
      imageCouverture:      json['image_couverture'],
      statut:               json['statut'],
    );
  }
}